.github/workflows/superset-frontend.yml

```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - "master"
       - "[0-9].[0-9]"
-  pull_request:
+  pull_request_target:
```

Comment:
as per docs:
```
Warning: For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. Additionally, any caches share the same scope as the base branch. To help prevent cache poisoning, you should not save the cache if there is a possibility that the cache contents were altered. For more information, see "[Keeping your GitHub Actions and workflows secure: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)" on the GitHub Security Lab website.
```

secrets can leak to forks, this trigger runs github workflow code from the base branch (master) but we are checking out code from the PR ref on these workflows, that can open other vectors for attacks, for example by changing code on `/scripts` that would get executed by the workflow.

Reply:
That's right, the person calling from the fork can't change the workflow itself (the one from the main branch runs), but if that workflow calls say a script (or evolves into calling one down the line), then they could extract the env vars and/or secrets specified on the main branch for that action. We should review our `pull_request_target ` and put them in a special place where we have extra caution.

Reply:
Yep... I'll audit more closely for risk here, fix what I can, and add comments on what doesn't need fixing. Hopefully we can add linting/enforcement after that so other people that don't understand these complexities don't make the same mistake.